woocommerce/woocommerce-gutenberg-products-block
# Update @woocommerce/components

Fixes #3603.
Fixes #4192.
Fixes #1243.
Fixes #1684.

This PR:
* Updates `@woocommerce/components` to version 6.2.0. That required making some changes here and there. They are explained in #3603.
* Creates a `ExpandableSearchListItem` component. There were several controls having a similar logic to make search list items expandable and it could be extracted to its own component. In the future, this could be uplifted to `@woocommerce/components`.
* I refactored and cleaned up some parts of the modified components.

#### Accessibility




### Screenshots

| Before | After |
| --- | --- |
| ![imatge](https://user-images.githubusercontent.com/3616980/118507132-3cb7f700-b72e-11eb-9372-540382329e96.png) | ![imatge](https://user-images.githubusercontent.com/3616980/118507204-4f323080-b72e-11eb-877c-905338e98986.png) |

### How to test the changes in this Pull Request:

The controls modified in this PR are used in many blocks, so the steps below should be repeated for all of them.

1. Add the block to a page and ensure you can select items in the controls.
2. If there are expandable items (Featured Products or Products by Attribute), make sure you can select inner items.
3. If a count is displayed, make sure it's displayed correctly.
4. Verify that you can do this actions also in the control which are in the sidebar:
![imatge](https://user-images.githubusercontent.com/3616980/115864612-63d51e80-a437-11eb-9553-db9f115081e4.png)
5. Test adding several blocks in the same page, verify controls work independently of each other.

Affected blocks:
* Featured Product Block
* Featured Category Block
* Hand-Picked products Block
* Products by Category Block
* Products by Tag Block
* Products by Attribute Block
* Reviews by Product
* Reviews by Category
* Filter Products by Attribute

### Changelog

> Improved accessibility and styling of the controls of several of ours blocks.
